src/modules/rover_pos_control/RoverPositionControl.cpp

```diff
@@ -346,15 +379,43 @@ RoverPositionControl::control_attitude(const vehicle_attitude_s &att, const vehi
 
 }
 
+void
+RoverPositionControl::control_rates(const vehicle_angular_velocity_s &rates, const matrix::Vector3f &current_velocity,
+				    const vehicle_rates_setpoint_s &rates_sp)
+{
+	struct ECL_ControlData control_input = {};
+	const float current_speed = current_velocity.norm();
+
+	// Set scaling factor with local velocity
+	control_input.groundspeed = current_speed;
+	// Lock integrator when local velocity is small
+	control_input.lock_integrator = (current_speed < _param_rate_i_minspeed.get());
+
+	control_input.body_z_rate = rates.xyz[2];
+	_att_control.set_bodyrate_setpoint(rates_sp.yaw);
+
+	float control_effort = _att_control.control_bodyrate(control_input);
+	control_effort = math::constrain(control_effort, -1.0f, 1.0f);
+
+	_act_controls.control[actuator_controls_s::INDEX_YAW] = control_effort;
+
+	const float control_throttle = rates_sp.thrust_body[0];
+
+	_act_controls.control[actuator_controls_s::INDEX_THROTTLE] =  math::constrain(control_throttle, 0.0f, 1.0f);
+}
+
+
 void
 RoverPositionControl::run()
 {
 	_control_mode_sub = orb_subscribe(ORB_ID(vehicle_control_mode));
 	_global_pos_sub = orb_subscribe(ORB_ID(vehicle_global_position));
 	_local_pos_sub = orb_subscribe(ORB_ID(vehicle_local_position));
+	_vehicle_angular_velocity_sub = orb_subscribe(ORB_ID(vehicle_angular_velocity));
```

Comment:
We should switch these to uORB::Subscription at some point.

Reply:
Right, I will do it on a separate PR